websockify/plugins.py

```diff
@@ -0,0 +1,47 @@
+from websockify.traffic_plugins import BasePlugin
+from websockify import pyDes
+
+class RFBDes(pyDes.des):
+    def setKey(self, key):
+        """RFB protocol for authentication requires client to encrypt
+           challenge sent by server with password using DES method. However,
+           bits in each byte of the password are put in reverse order before
+           using it as encryption key."""
+        newkey = []
+        for ki in range(len(key)):
+            bsrc = ord(key[ki])
+            btgt = 0
+            for i in range(8):
+                if bsrc & (1 << i):
+                    btgt = btgt | (1 << 7 - i)
+            newkey.append(btgt)
+        super(RFBDes, self).setKey(newkey)
+
+"""Intercepts VNC traffic and injects the password from a provided token"""
+class VncTokenAuthenticationTrafficPlugin(BasePlugin):
+    def __init__(self, handler, tsock):
+        super().__init__(handler, tsock)
+        self.password = self.handler.target_attribtues["password"]
+        self.client_packet_count = 0
+        self.target_packet_count = 0
+
+    def from_client(self, s):
+        self.client_packet_count += 1
+        if self.client_packet_count == 2:
+            return None
+        return s
+
+    def from_target(self, s):
+        self.target_packet_count += 1
+        if self.target_packet_count == 2 and b"\x02" in s[1:]: # check if password is supported
```

Comment:
I'm afraid this is a bit fragile. There is no guarantee that the stream will be split in packages in a specific way.

Some rudimentary protocol handling will be needed here.

Have you considered a proxy model, instead of this filtering model? It might be a better fit.